### PR TITLE
contexts: remove all TODO instances

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: internal/consensus/common_test.go [hunk 1]
```diff
@@ -189,34 +190,22 @@ func incrementRound(vss ...*validatorStub) {
...
+		}
+		vssj, err := vss[j].GetPubKey(ctx)
+		if err != nil {
+			panic(err)
+		}
 
-func (vss ValidatorStubsByPower) Swap(i, j int) {
-	it := vss[i]
-	vss[i] = vss[j]
-	vss[i].Index = int32(i)
```
Comment (creachadair) [inline internal/consensus/common_test.go:217]: I think we may have to preserve this—the default swap is a by-value exchange, and won't update the embedded field.

Comment (tychoish) [reply internal/consensus/common_test.go]: oh god, this cased everything to blow up and I kept looking at the less  function and swap was the thing that did it. :headdesk:

File: internal/consensus/common_test.go [hunk 2]
```diff
@@ -190,22 +189,36 @@ func incrementRound(vss ...*validatorStub) {
 	}
 }
 
-func sortVValidatorStubsByPower(ctx context.Context, vss []*validatorStub) {
-	sort.Slice(vss, func(i, j int) bool {
-		vssi, err := vss[i].GetPubKey(ctx)
+type ValidatorStubsByPower []*validatorStub
```
Comment (creachadair) [inline internal/consensus/common_test.go:193]: If you want, you can still inject a context:

```go
type byPower struct{
   ctx  context.Context
   vals []*Whatever
}

func (b byPower) Len() int { return len(b.vals) }
func (b byPower) Swap(i, j int) {
    // do stuff with b.ctx and b.vals to swap
}
// etc.
```

Comment (tychoish) [reply internal/consensus/common_test.go:193]: that's fair, I think my method also works, and avoids needing to put a context in a struct.

**Review by creachadair — APPROVED** [review]

**Pull request merged**